third_party/gtest/CMakeLists.txt

```diff
@@ -1,21 +1,27 @@
-message(STATUS "Configuring and building gtest")
-if(MSVC)
-    # Force using shared runtime library when MSVC builds shared libraries
-    ginkgo_load_git_package(gtest_external
-        "https://github.com/google/googletest.git"
-        "6a7ed316a5cdc07b6d26362c90770787513822d4"
-        # Work around the linking errors when compiling gtest with CUDA
-        "-Dgtest_disable_pthreads=ON" "-Dgtest_force_shared_crt=ON")
-else()
-    ginkgo_load_git_package(gtest_external
-        "https://github.com/google/googletest.git"
-        "6a7ed316a5cdc07b6d26362c90770787513822d4"
-        # Work around the linking errors when compiling gtest with CUDA
-        "-Dgtest_disable_pthreads=ON"
-        "-DCMAKE_CXX_FLAGS=-fPIC")
+if(NOT GTest_FOUND)
+    message(STATUS "Fetching external GTest")
+    include(FetchContent)
+    FetchContent_Declare(
+        googletest
+        GIT_REPOSITORY https://github.com/google/googletest.git
+        GIT_TAG        6a7ed316a5cdc07b6d26362c90770787513822d4
+    )
+    # need to set the variables in CACHE due to CMP0077
+    set(gtest_disable_pthreads ON CACHE INTERNAL "")
+    if(MSVC)
+        set(gtest_force_shared_crt ON CACHE INTERNAL "")
+    endif()
+    FetchContent_GetProperties(googletest)
+    if(NOT googletest_POPULATED)
+        FetchContent_Populate(googletest)
+    endif()
+    add_subdirectory(${googletest_SOURCE_DIR} ${googletest_BINARY_DIR} EXCLUDE_FROM_ALL)
+    # make sure the tests DLLs are placed in the working path for CTest
+    set_target_properties(gtest gtest_main PROPERTIES
+        RUNTIME_OUTPUT_DIRECTORY "${GINKGO_LIBRARY_PATH}"
+        ARCHIVE_OUTPUT_DIRECTORY "${GINKGO_LIBRARY_PATH}"
+        LIBRARY_OUTPUT_DIRECTORY "${GINKGO_LIBRARY_PATH}")
```

Comment:
Is this for Windows specifically then? Or is there some RPATH issue on linux as well?

Reply:
see my explanation for test WORKING_DIRECTORY above